src/Cake.Common/Text/TextTransformation.cs

```diff
@@ -64,7 +64,7 @@ public void Save(FilePath path)
             // Render the content to the file.
             var file = _fileSystem.GetFile(path);
             using (var stream = file.OpenWrite())
-            using (var writer = new StreamWriter(stream, Encoding.UTF8, 1024, true))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false), 1024, true))
```

Comment:
not exactly... the `false` param tells `StreamWriter` NOT to emit the BOM when writing the stream.
A `Stream` itself never implicitly has a BOM (ie. you won't see the BOM in the stream's bytes no matter what). The BOM comes from the `Encoding`'s specific `GetPreamble` method.

Strangely, when you use `Encoding.UTF8`, it uses the constructor that explicitly ADDS the preamble (BOM): http://referencesource.microsoft.com/#mscorlib/system/text/encoding.cs,1547.

But the DEFAULT constructor for `UTF8Encoding` explicitly DOES NOT add the BOM: http://referencesource.microsoft.com/#mscorlib/system/text/utf8encoding.cs,65.

In my opinion, you'll have more success across various platforms if you don't add the BOM to the written file in this case.